src/main/java/io/r2dbc/h2/codecs/DefaultCodecs.java

```diff
@@ -32,7 +34,7 @@
     private final List<Codec<?>> codecs;
 
     public DefaultCodecs(Client client) {
-        this.codecs = Arrays.asList(
+        this.codecs = new ArrayList<>(Arrays.asList(
             new BigDecimalCodec(),
```

Comment:
Since `new ArrayList` is exactly what `Arrays.asList` does, there is no need to wrap in yet another list.

Reply:
Absolutely, Good Catch, Thanks!

Reply:
Now I remembered that i did this in order to get a modifiable list, but in the new commit I took a better approach in order to keep (i guess) the original intention, **fixed-size list backed by the specified array** (as `Array.asList` does). 
Thanks!